src/Admin/TemplateRegistryInterface.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\Admin;
+
+interface TemplateRegistryInterface
+{
+    public function __construct(array $templates);
```

Comment:
A constructor should never be part of an interface I think.

Reply:
Should I remove the constructor from the interface or re-add the `setTemplates` function?

Reply:
Just remove the constructor seems fine.

Reply:
I removed the constructor @greg0ire